tests/recorder_snapshot_test.cc

```diff
@@ -13,8 +13,11 @@
 
 #include <algorithm>
 #include <chrono>
+#include <cstddef>
 #include <filesystem>
 #include <memory>
+#include <mutex>
+#include <ratio>
```

Comment:
ratio is only required when using ratio directly, e.g. milli.
milliseconds is defined by chrono using ratio, but not in ratio.